src/datumaro/plugins/data_formats/ade20k2017.py

```diff
@@ -108,9 +107,10 @@ def _load_items(self, subset):
                     attributes = {k: True for k in v["attributes"]}
 
                     item_annotations.append(
-                        Mask(
+                        ExtractedMask(
+                            index_mask=mask,
+                            index=instance_id,
```

Comment:
If so, there might be the copied `index_mask`s for all annotations in an item. For instance, if an image is composed of 100-class indices, we should cope an index mask for 100 times. Don't we have any memory leakage problem?

Reply:
No, we do not copy it. `index_mask` is a reference (pointer) of where the mask data (2d array) exists.  We do this to maintain a single source "explicitly" if multiple binary masks (e.g., `datumaro.Mask`) are created from the single source.

The advantage of explicitly maintaining `index_mask` reference is described in the PR description above:
> | | |
> | :-: | :-: |
> | After | 1) Check all Mask annotations are ExtractedMask and it share the same index mask, 2) Just extract the source index mask |

If we want to merge multiple `ExtractedMask` from the same source into one, we can simply bring the `index_mask` rather than merging each binary mask step by step.

Reply:
Got it. If so, all masks from an image should have the same index. It might be less possible, but you considered the case that an image has multiple class index masks (for instance, if an image can have multiple label schema for the segmentation). Is this true?

Reply:
I guess I cannot understand your comment well. Can we proceed this conversation in Teams?